EventEmitter.d.ts

```diff
@@ -19,30 +19,30 @@ declare namespace EventEmitter {
         [event: string]: Listener[];
     }
 
-    type Listeners = Listener[]|ListenerMap;
+    type Listeners = Listener[] | ListenerMap;
 
-    export class EventEmitter {
+    export class EventEmitter<E = EventKey> {
         static noConflict(): typeof EventEmitter;
 
-        getListeners(event: EventKey): Listeners;
+        getListeners(event: E): Listeners;
         flattenListeners(listeners: Listener[]): void;
-        getListenersAsObject(event: EventKey): ListenerMap;
-        addListener(event: EventKey, listener: Listener|Function): this;
-        on(event: EventKey, listener: Listener|Function): this;
-        addOnceListener(event: EventKey, listener: Function): this;
-        once(event: EventKey, listener: Function): this;
-        defineEvent(event: EventKey): this;
-        defineEvents(events: EventKey[]): this;
-        removeListener(event: EventKey, listener: Function): this;
-        off(event: EventKey, listener: Function): this;
+        getListenersAsObject(event: E): ListenerMap;
+        addListener(event: E, listener: Listener | Function): this;
+        on(event: E, listener: Listener | Function): this;
+        addOnceListener(event: E, listener: Function): this;
+        once(event: E, listener: Function): this;
+        defineEvent(event: E): this;
+        defineEvents(events: E[]): this;
+        removeListener(event: E, listener: Function): this;
+        off(event: E, listener: Function): this;
         addListeners(event: Events, listeners: Function[]): this;
         removeListeners(event: Events, listeners: Function[]): this;
         manipulateListeners(remove: boolean, event: Events, listeners: Function[]): this;
-        removeEvent(event?: EventKey): this;
-        removeAllListeners(event?: EventKey): this;
-        emitEvent(event: EventKey, args?: any[]): this;
-        trigger(event: EventKey, args?: any[]): this;
-        emit(event: EventKey): this;
```

Comment:
This would not throw an exception but it would thorow an error while trying to compile (or transpile) typescript. js code it not infected beaause the .d.ts file it not loaded.

the added functionallity meight be the folllowing:

```
export enum Events {
   CoolEvent, EvenCoolerEvent
}

export class CoolClass extends EventEmitter<Events> {

   constructor() {
      this.emit(Events.CoolEvent); // this meight work because of the enum type in generic
  }
}
```

currently if i try to extends EventEmitter the required param of `emit()` or other function is `EventKey` and EventKey is not an Enum.

So you now can type whateher type you whant as `EventKey`.

You need to know that Typescript handles Enumerations as numeric value to transpile down to javascript. That means if you have an enum `enum TestEnum { cool}`  and you type `TestEnum.cool` it will result in a numeric value of 0 because Typescript uses an object with an string index to find out the value (wich is numeric). so an enum ist not a string or RegExp.

I'll hope this helped!